src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.hpp

```diff
@@ -130,81 +141,123 @@ class FlightTaskAutoFollowTarget : public FlightTask
-	matrix::Vector3f calculate_drone_desired_position(matrix::Vector3f target_position, matrix::Vector2f offset_vector);
+	/**
+	 * Publishes gimbal control command to track the target, given xy distance and z (height) difference.
+	 */
+	void point_gimbal_at(float xy_distance, float z_distance);
 
-	TargetEstimator _target_estimator;
+	/**
+	 * Get the current follow-me perspective setting from PX4 parameters
+	 *
+	 * @param param_nav_ft_fs value of the parameter NAV_FT_FS
+	 * @return Angle [deg] from which the drone should view the target while following it, with zero degrees indicating the target's 12 o'clock
+	 */
+	float get_follow_me_angle_setting(int param_nav_ft_fs) const;
 
-	// Follow angle is defined with 0 degrees following from front, and then clockwise rotation
-	float _follow_angle_rad{0.0f};
-	AlphaFilter<float> _follow_angle_filter;
+	// Sticks object to read in stick commands from the user
+	Sticks _sticks;
 
 	// Estimator for target position and velocity
+	TargetEstimator _target_estimator;
 	follow_target_estimator_s _follow_target_estimator;
-	matrix::Vector2f _target_velocity_unit_vector;
 
-	// Lowpass filters for smoothingtarget position because it's used for setpoint generation
-	AlphaFilter<matrix::Vector3f> _target_position_filter;
+	// Last target estimator timestamp to handle timeout filter reset
+	uint64_t _last_valid_target_estimator_timestamp{0};
+
+	// Second Order Filter to calculate kinematically feasible target position
+	SecondOrderReferenceModel<matrix::Vector3f> _target_pose_filter;
+
+	// Internally tracked Follow Target characteristics, to allow RC control input adjustments
+	float _follow_target_distance{0.0f};
+	float _follow_target_height{0.0f};
+	float _follow_angle_rad{0.0f}; // 0 degrees following from front, and then clockwise rotation
+
+	// Estimated (Filtered) target orientation setpoint
+	float _target_orientation_rad{0.0f};
+
+	// Orbit angle setpoint, measured in global frame, against the target
+	float _orbit_angle_setpoint{0.0f};
+
+	// Actual drone to target heading [rad]
+	float _drone_to_target_heading{0.0f};
+
+	// Actual orbit angle in relation to the target
+	float _measured_orbit_angle{0.0f};
 
-	// Lowpass filter for smoothing the offset vector and have more dynamic shots when target changes direction
-	AlphaFilter<matrix::Vector2f> _offset_vector_filter;
+	// Actual drone to target 2d position vector
+	Vector2f _drone_to_target_vector{0.0f, 0.0f};
 
-	// Lowpass filter assuming  values 0-1, for avoiding big steps in velocity feedforward
-	AlphaFilter<float> _velocity_ff_scale;
+	// Tracked orbit tangential speed, to compensate for the orbital motion for velocity setpoints
+	Vector2f _orbit_tangential_velocity{0.0f, 0.0f};
 
 	// NOTE: If more of these internal state variables come into existence, it
 	// would make sense to create an internal state machine with a single enum
 	bool _emergency_ascent = false;
```

Comment:
already commented as well in cpp file .. but again - why does this need to be an internal state? can we delete?